README.rst

```diff
@@ -79,9 +79,9 @@ Options
 
 * ``awslogs groups``: List existing groups
 * ``awslogs streams GROUP``: List existing streams withing ``GROUP``
-* ``awslogs get GROUP [STREAM_EXPRESSION]``: Get logs matching ``STREAM_EXPRESSION`` in ``GROUP``.
+* ``awslogs get GROUP [STREAM_EXPRESSION]``: Get logs from streams with names matching ``STREAM_EXPRESSION`` in log group ``GROUP``.
 
-  - Expressions can be regular expressions or the wildcard ``ALL`` if you want any and don't want to type ``.*``.
+  - STREAM_EXPRESSION is a python regular expression accepted by ``re.compile()`` `described here <https://docs.python.org/3/library/re.html#regular-expression-syntax>`_. Expression ``ALL`` is reserved and is same as ``'.*'``. Remember to quote/escape shell special characters to ensure they are not gobbled up by shell variable expansion. E.g. ``'2014-04.*'`` instead of ``2014-04.*``
```

Comment:
This doesn't mention how the regex is used. It is used in [`.match(…)`](https://docs.python.org/3/library/re.html#re.Pattern.match), rather than say [.fullmatch(…)](https://docs.python.org/3/library/re.html#re.Pattern.fullmatch) (which I might prefer but have no strong opinion†) ,or  [`.search(…)`](https://docs.python.org/3/library/re.html#re.Pattern.search) (which it sounds like that's how people imagine it being used).

† changing from `.match(…)` would be a breaking change (dropping the `^` but still using `.match(…)` is not as it acts exactly the same) so should be a major version change+in release notes